### ISSUE TITLE
Dependency version conflicts

### ISSUE THREAD

**Issue by radioflash** [sisl/GaussianFilters.jl#32]
The Julia package registry implicitly added some [compat] information for all the dependencies of this package.
But those are now out of date (but everything just works when circumventing those compat checks).

This is a problem because these implicit dependencies are practically invisible and this leads to version conflicts 
that are *really* annoying to debug, especially for newer users.

From what I found out, the Julia package registry made compat information mandatory for all dependencies (meaning it can't just be removed there).

IMO, the best way to solve this would be to adapt/update the compat info from the registry and integrate it directly in Project.toml, because this would be much more transparent for users and solve current version conflicts!

Alternatively, the "hidden" compat info that the Julia package registry provides for 
GaussianFilters.jl could also be changed without touching this project.

TL;DR:
I could do either:
- pull request for this projects project.toml (preferable)
- pull request for the registry

Are pull request for additional filter implementations or API extensions acceptable in general (because I have some ideas, but I'm unsure if this project is still live)?

Also, thanks for the amazing work. This project was already super useful to me!

**Comment by lassepe** [comment]
I think it is a good idea to add the compat entries here. I noticed that quite a few package suffer from this issue since Pkg started auto-adding compat entries in the registry. cc @zsunberg or @MaximeBouton

**Comment by zsunberg** [comment]
Hi @radioflash ,

Thanks for your interest in contributing. Yes, I think it is best to add compat entries to the Project.toml

> Are pull request for additional filter implementations or API extensions acceptable in general (because I have some ideas, but I'm unsure if this project is still live)?

It seems like this repo has gone somewhat stale - it could use some fresh life! That being said, adding API restrictions can be very tricky, so please request some reviews! @lassepe in particular is very good about giving helpful reviews and designing these interfaces.

@jamgochiana @duncaneddy , it looks like y'all were the original contributors. Did you have any plans for the future of this project?
